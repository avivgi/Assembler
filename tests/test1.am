.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
prn #-5
mov STR[5], STR[2]
sub r1, r4
cmp r4, #sz
bne END
L1: bne LOOP
END: hlt
.define len = 4
STR: .string "abcdef"
LIST: .data 6, -9, len
cmp r6, #sz
cmp r7, #sz
cmp r6, #sz
cmp r7, #sz
K: .data 22
