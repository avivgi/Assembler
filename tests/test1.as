.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
mcr m_mcr1
cmp r4, #sz
bne END
endmcr
prn #-5
mov STR[5], STR[2]
sub r1, r4
m_mcr1
L1: bne LOOP
END: hlt
.define len = 4
mcr m_mcr2
cmp r6, #sz
cmp r7, #sz
endmcr
STR: .string "abcdef"
LIST: .data 6, -9, len
K: .data 22
m_mcr2
m_mcr2
K: .data 22
