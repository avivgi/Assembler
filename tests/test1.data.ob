2 12
-2080803884 00000000000000 **************
0101 00001111000000 ****####******
0102 00000001100001 *******##****#
0103 00000001100010 *******##***#*
0104 00000001100011 *******##***##
0105 00000001100100 *******##**#**
0106 00000001100101 *******##**#*#
0107 00000001100110 *******##**##*
0107 00000000000000 **************
0109 00000000000110 ***********##*
0110 11111111110111 ##########*###
0111 00000000000100 ***********#**
0112 00000000010110 *********#*##*
0113 00000000010110 *********#*##*
