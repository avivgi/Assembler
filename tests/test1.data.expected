Address: 135	value: 97	 word: 00000001100001
Address: 136	value: 98	 word: 00000001100010
Address: 137	value: 99	 word: 00000001100011
Address: 138	value: 100	 word: 00000001100100
Address: 139	value: 101	 word: 00000001100101
Address: 140	value: 102	 word: 00000001100110
Address: 141	value: 0	 word: 00000000000000
Address: 142	value: 6	 word: 00000000000110
Address: 143	value: -9	 word: 11111111110111
Address: 144	value: 4	 word: 00000000000100
Address: 145	value: 22	 word: 00000000010110
